### PR TITLE
Slightly clarify the implementation of safe_masked_invalid.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/cbook/__init__.py [hunk 1]
```diff
@@ -746,12 +746,9 @@ def safezip(*args):
...
     if not x.dtype.isnative:
-        # Note that the argument to `byteswap` is 'inplace',
-        # thus if we have already made a copy, do the byteswap in
-        # place, else make a copy with the byte order swapped.
-        # Be explicit that we are swapping the byte order of the dtype
-        x = x.byteswap(copy).newbyteorder('S')
-
+        # If we have already made a copy, do the byteswap in place, else make a
+        # copy with the byte order swapped.
+        x = x.byteswap(inplace=copy).newbyteorder('native')
```
Comment (timhoffm) [inline lib/matplotlib/cbook/__init__.py]: Where is 'native' documented? (https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.newbyteorder.html)

Comment (anntzer) [reply lib/matplotlib/cbook/__init__.py]: > The code does a case-insensitive check on the first letter of new_order for the alternatives above. For example, any of ‘B’ or ‘b’ or ‘biggish’ are valid to specify big-endian.

So it's not *explicitly* documented but I think writing "native" is clearer for someone who doesn't have the ref doc under their eyes right now.

Comment (timhoffm) [reply lib/matplotlib/cbook/__init__.py]: I don't trust undocumented behavior. Numpy might change that without warning. I'd rather go for

```suggestion
        x = x.byteswap(inplace=copy).newbyteorder('N')  # native byte order
```

Comment (timhoffm) [reply lib/matplotlib/cbook/__init__.py]: Oh, IMHO that's not a good API on numpy's side (I could write `newbyteorder("no change")`, which would still convert to native). I wouldn't wonder if they'd narrow it down at some point. I'd still go with the explicitly mentioned value like this:

```suggestion
        x = x.byteswap(inplace=copy).newbyteorder('N')  # native byte order
```

Comment (anntzer) [reply lib/matplotlib/cbook/__init__.py]: done

**Pull request merged**